singleton-service/src/service/deltas.rs

```diff
@@ -1,26 +1,99 @@
 use chrono::offset::Utc;
 use chrono::DateTime;
-use std::cell::RefCell;
 use std::collections::HashMap;
 use threescale::structs::ThreescaleData;
 
-// getter and setter vs dot
+#[derive(Clone)]
+pub struct AppDelta {
+    pub key_type: String,
+    pub usages: HashMap<String, u32>,
+}
```

Comment:
@unleashed yeah it makes sense to have a separate type to handle different types of applications. If I implement that way, then there's no point of having a separate `AppDelta` struct because it contains only hits. So I can have something like below for Delta store. 

```rust

pub enum AppType {
    AppId,
    AppIdAppKey,
    UserKey,
    OAuthToken
}

pub struct App {
    pub type: AppType,
    pub user_key: Some(String),
    pub app_id: Some(String),
    pub oauth_token: Some(String)
}

pub struct DeltaStore {
    pub last_update: Option<DateTime<Utc>>,
    pub deltas: HashMap<String, HashMap<App, HashMap<String, u32>>>,
    pub request_count: u32,
    pub capacity: u32,
}

```
Want to clarify this first because this will change several places.

Reply:
@unleashed Also we need this type for app identifier in both cache filter and singleton because we need to have a common and uniform way to have the app identifier across cache and singleton. This will be implemented from rahul's PR when he sends me data from the cache filter. For this PR I'll keep the structs as they are now and will change after rahul's PR.

Reply:
Yes, I think that's how it would look like.